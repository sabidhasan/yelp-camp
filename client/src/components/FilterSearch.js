import React from 'react'
import Select from './Select'
import InputRange from 'react-input-range';
import MultiCheckBox from './MultiCheckBox'
import 'react-input-range/lib/css/index.css';

class FilterSearch extends React.Component {
  constructor(props) {
    super(props);
    // console.log(props);
    this.state = {
      selectedProv: null,
      selectedActivities: props.filterCriteria.activities,
      selectedRegions: props.filterCriteria.regions,
      selectedDistances: {min: 0, max: props.filterCriteria.maxDistance},
      sortBy: null,
      filterAreaExpanded: false,
    }
    this.handleChange = this.handleChange.bind(this);
    this.toggleExpanded = this.toggleExpanded.bind(this);
  }

  toggleExpanded() {
    this.setState({filterAreaExpanded: !this.state.filterAreaExpanded})
  }

  handleChange(newValue, property) {
    // Apply filter
    this.setState({[property]: newValue}, () => {
      if (this.props.onChange) this.props.onChange(this.state)
    });
  }

  render() {
    return (
      <div className={`FilterSearch ${this.props.className}`}>
        {/* <h2>Filters</h2> */}
        <button
          className='FilterSearch__button btn bold btn--flat'
          onClick={this.toggleExpanded}>
          <i class="fas fa-sliders-h"></i>
          {this.state.filterAreaExpanded ? 'Hide ' : 'Show '} Filters
        </button>
        <div className={`FilterSearch__tiles ${this.state.filterAreaExpanded ? 'FilterSearch__tiles--expanded' : ''}`}>
          <div className='FilterSearch__section'>
            <h2>Region</h2>
            {this.props.filterCriteria.regions.length ?
              <MultiCheckBox
                items={this.props.filterCriteria.regions}
                defaultText={'Select Regions'}
                onChange={val => this.handleChange(val, 'selectedRegions')}
              />
            : null }
          </div>
          <div className='FilterSearch__section FilterSearch__distance'>
            {this.props.filterCriteria.maxDistance > 0 ?
              <React.Fragment>
                <h2>Distance</h2>
                <InputRange
                  minValue={0}
                  maxValue={this.props.filterCriteria.maxDistance}
                  formatLabel={value => `${value}km`}
                  value={this.state.selectedDistances}
                  onChange={val => {
                    console.log(val);
                    this.setState({selectedDistances: val})

                  }}
                  onChangeComplete={val => this.handleChange(val, 'selectedDistances')}
                />
              </React.Fragment>
            : null}
            <h2>Province</h2>
            <Select
              items={this.props.filterCriteria.provinces}
              defaultText={'All Provinces'}
              onChange={val => this.handleChange(val, 'selectedProv')}
            />

            <h2>Sort</h2>
            <Select
              items={[{text: 'Distance', value: 'Distance'}]}
              defaultText={'Best Match'}
              onChange={val => this.handleChange(val, 'sortBy')}
            />

          </div>
          <div className='FilterSearch__section'>
            <h2>Activities</h2>
            {this.props.filterCriteria.activities.length ?
              <MultiCheckBox
                items={this.props.filterCriteria.activities}
                defaultText={'Select Activities'}
                onChange={val => this.handleChange(val, 'selectedActivities')}
              />
            : null }
          </div>
        </div>
      </div>
    )
  }
}

export default FilterSearch
