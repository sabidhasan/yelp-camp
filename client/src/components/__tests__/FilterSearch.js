import React from 'react'
import toJson from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import FilterSearch from '../FilterSearch'


configure({ adapter: new Adapter() });

describe('<MultiCheckBox />', () => {
  let component;
  let props;
  const onChange = jest.fn();
  const className = 'Test-Class'

  beforeEach(() => {
    props = {
      onChange,
      className,
      filterCriteria: {
        maxDistance: 2,
        regions: [1, 2, 3, 4],
        provinces: [1, 2, 3, 4],
        activities: [1, 2, 3, 4]
      },
      items: ['First', 'Second', 'Third'],
      defaultText: 'Test Text',
    }
    component = shallow(<FilterSearch { ...props } />)
  })

  it('Component should be defined', () => {
    expect(component).toBeDefined();
  });

  it('Prop classname is added to DOM element', () => {
    expect(component.find(`.${className}`)).toHaveLength(1);
  });

  it('Button text says "show"', () => {
    expect(component.find('.FilterSearch__button').text()).toBe('Show Filters');
  });

  it('When clicked, button text says "hide"', () => {
    component.find('.FilterSearch__button').simulate('click');
    expect(component.find('.FilterSearch__button').text()).toBe('Hide Filters');
  });

  it('When clicked, button text says "hide"', () => {
    component.find('.FilterSearch__button').simulate('click');
    expect(component.find('.FilterSearch__tiles--expanded')).toHaveLength(1);
  });

  it('When maxDistance is greater than 0, show InputRange', () => {
    expect(component.find('InputRange')).toHaveLength(1);
  });

  it('When maxDistance is equal to 0, do not show InputRange', () => {
    props = {
      onChange,
      filterCriteria: {
        maxDistance: 0, regions: [1, 2, 3, 4],
        provinces: [1, 2, 3, 4], activities: [1, 2, 3, 4]
      }
    }
    component = shallow(<FilterSearch { ...props } />)
    expect(component.find('InputRange')).toHaveLength(0);
  });

  it('LoginForm component snapshot matches what is expected', () => {
    expect(toJson(component)).toMatchSnapshot();
  })
});
