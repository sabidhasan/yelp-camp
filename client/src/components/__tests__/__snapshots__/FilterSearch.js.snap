// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiCheckBox /> LoginForm component snapshot matches what is expected 1`] = `
<div
  className="FilterSearch Test-Class"
>
  <button
    className="FilterSearch__button btn bold btn--flat"
    onClick={[Function]}
  >
    <i
      className="fas fa-sliders-h"
    />
    Show
     Filters
  </button>
  <div
    className="FilterSearch__tiles "
  >
    <div
      className="FilterSearch__section"
      tabIndex="0"
    >
      <h2>
        Region
      </h2>
      <MultiCheckBox
        defaultText="Select Regions"
        items={
          Array [
            1,
            2,
            3,
            4,
          ]
        }
        onChange={[Function]}
      />
    </div>
    <div
      className="FilterSearch__section FilterSearch__distance"
    >
      <h2>
        Distance
      </h2>
      <InputRange
        allowSameValues={false}
        classNames={
          Object {
            "activeTrack": "input-range__track input-range__track--active",
            "disabledInputRange": "input-range input-range--disabled",
            "inputRange": "input-range",
            "labelContainer": "input-range__label-container",
            "maxLabel": "input-range__label input-range__label--max",
            "minLabel": "input-range__label input-range__label--min",
            "slider": "input-range__slider",
            "sliderContainer": "input-range__slider-container",
            "track": "input-range__track input-range__track--background",
            "valueLabel": "input-range__label input-range__label--value",
          }
        }
        disabled={false}
        formatLabel={[Function]}
        maxValue={2}
        minValue={0}
        onChange={[Function]}
        onChangeComplete={[Function]}
        step={1}
        value={
          Object {
            "max": 2,
            "min": 0,
          }
        }
      />
      <h2>
        Province
      </h2>
      <Select
        defaultText="All Provinces"
        items={
          Array [
            1,
            2,
            3,
            4,
          ]
        }
        onChange={[Function]}
      />
      <h2>
        Sort
      </h2>
      <Select
        defaultText="Best Match"
        items={
          Array [
            Object {
              "text": "Distance",
              "value": "Distance",
            },
          ]
        }
        onChange={[Function]}
      />
    </div>
    <div
      className="FilterSearch__section"
    >
      <h2>
        Activities
      </h2>
      <MultiCheckBox
        defaultText="Select Activities"
        items={
          Array [
            1,
            2,
            3,
            4,
          ]
        }
        onChange={[Function]}
      />
    </div>
  </div>
</div>
`;
